### PR TITLE
Added vendor-update command to makefile

### PR DESCRIPTION
**Pull request by surajnarwade**
This PR adds vendor-update command to Make, so that
we can simply update vendor using `make vendor-update` command

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -98,3 +98,8 @@ packages:
 .PHONY: upload-packages
 upload-packages:
 	./scripts/upload-packages.sh
+
+# Update vendoring
+.PHONY: vendor-update
+vendor-update:
+	glide update --strip-vendor
```
Comment (cdrage) [inline Makefile]: shouldn't we add `glide update -v` before too?

Comment (surajnarwade) [reply Makefile]: According to odo doc, this should be sufficient I guess.
I was thinking of adding `glide-vc --only-code --no-tests` as well 
WDYT ?

Comment (cdrage) [reply Makefile]: I don't think we're adding that anymore. Using `glide-vc` and all.

Maybe we should echo after vendor-update saying that the user should `git add .` and do a separate commit? Similar to how `developers.md` describes it.

Comment (surajnarwade) [reply Makefile]: using `glide-vc` removes extra clutter which solves https://github.com/redhat-developer/odo/issues/670 as well

Comment (cdrage) [reply Makefile]: I believe there's a reason why we don't use glide-vc, @kadel would know once he gets back.

Comment (surajnarwade) [reply Makefile]: Got it, so as of now, we can merge this in

Comment (cdrage) [reply Makefile]: agreed. this LGTM.

**Pull request merged**